pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -323,7 +323,7 @@ func (cb *ClusterBuilder) buildCluster(name string, discoveryType cluster.Cluste
 				c.DnsLookupFamily = cluster.Cluster_V4_ONLY
 			}
 		}
-		c.DnsRefreshRate = cb.req.Push.Mesh.DnsRefreshRate
+		c.DnsRefreshRate = proto.Clone(cb.req.Push.Mesh.DnsRefreshRate).(*durationpb.Duration)
```

Comment:
we donot need clone, a shallow copy is ok 
```suggestion
		c.DnsRefreshRate = &(*cb.req.Push.Mesh.DnsRefreshRate)
```

Reply:
I am not sure it is actually, the internal state

```
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields
```

will cause issues I think? Or maybe that 'why' is wrong, but it still crashes with that.

Reply:
I've made some changes:
* Use a shallowcopy safe for protobuf
* Change other usages
* Refactor the clone/shallow clone to be more ergonomic